### ISSUE TITLE
Get-AzureRmResource can't handle Microsoft.Insights/Components/ExportConfiguration

### ISSUE THREAD

**Issue by gregjhogan** [Azure/azure-powershell#2405]
Trying to put something together that takes a desired state for app-insights continuous export and applies it.  Looks like to accomplish this you have to first query what is already set up for continuous export, and Get-AzureRmResource can't seem to handle what comes back.  Perhaps this isn't considered a resource?  If not, surely there is some other generic Azure powerhsell cmdlet that would support this (so I don't have to manually acquire and attach an auth token to a request)?  Thanks!

``` ps1
Get-AzureRmResource -Id '/subscriptions/{subscription-id}/{resource-group}/providers/microsoft.insights/components/{ai-name}/ExportConfiguration'
```

```
Get-AzureRmResource : Unable to cast object of type 'Newtonsoft.Json.Linq.JArray' to type 'Newtonsoft.Json.Linq.JObject'.
At line:1 char:1
+ Get-AzureRmResource -Id '/subscriptions/{subscription-id} ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-AzureRmResource], InvalidCastException
    + FullyQualifiedErrorId : Unable to cast object of type 'Newtonsoft.Json.Linq.JArray' to type 'Newtonsoft.
   Json.Linq.JObject'.,Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.GetAzureResourceCmdlet
```

**Comment by markcowl** [comment]
@ravbhatnagar Can you take a look?

**Comment by ravbhatnagar** [comment]
@gregjhogan Is this still an issue? This does not seem to be a resource type in the microsoft.insights namespace

**Comment by gregjhogan** [comment]
@ravbhatnagar @markcowl yes still an issue.  There doesn't seem to be support for setting the continuous export settings via an ARM template, so I am trying to build something that can get the continuous export settings to a desired state.  This requires the ability to read the current continuous export settings.  I don't know what exactly you consider a resource type, but you can certainly make a REST API call to `/subscriptions/{subscription-id}/{resource-group}/providers/microsoft.insights/components/{ai-name}/ExportConfiguration` and it returns an array of continuous export settings.  What do you think?

**Comment by cormacpayne** [comment]
This should be fixed in PR https://github.com/Azure/azure-powershell/pull/6008

**Comment by cormacpayne** [comment]
The fix for this should be available on May 7th in version 6.0.0 of `AzureRM` (version 6.0.0 of `AzureRM.Resources`). Please re-open this issue if the problem persists.
